### PR TITLE
Update error logging

### PR DESCRIPTION
**Pull request by perjor**
Guess I also sneaked in a filename change 😨 

Screenshot 2020-10-20 at 13 26 37

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -17,15 +17,7 @@
...
-    [
-      "@babel/plugin-transform-runtime",
-      {
-        "corejs": false,
-        "helpers": false,
-        "regenerator": false,
-        "useESModules": false
-      }
-    ],
+    "@babel/plugin-transform-runtime",
```
Comment (perprorenata) [inline .babelrc:20]: Are we sure we don't need these settings?

File: README.md [hunk 2]
```diff
@@ -79,19 +78,87 @@ errorHandler: (err) => console.log('VRR error, logging to the console since no h
...
+Example from VRR Test:
+
+```js
+try {
+  await seenHintsResource.get(null, id);
+} catch (error) {
+  expect(error.internalError.response.status).toBe(403);
+}
+```
+
```
Comment (perprorenata) [inline README.md:153]: 👍

File: package.json [hunk 3]
```diff
@@ -20,6 +20,10 @@
     {
       "name": "Graham Fairweather",
       "email": "graham.fairweather@prorenata.se"
+    },
+    {
+      "name": "Jordy Pereira",
+      "email": "jordy.pereira@prorenata.se"
     }
```
Comment (perprorenata) [inline package.json:27]: 😎

File: src/Rest.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+        clearTimeout(slowRequest);
+        clearTimeout(requestTimeout);
+        this.unregister(request);
+
+        const updated = {
+          ...request,
+          completed: Date.now(),
+          response: err.response && err.response.data,
+          status: 'failed',
+          internalError: err,
```
Comment (perprorenata) [inline src/Rest.js:206]: 👍

File: test-jest/methods.test.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    const seenHintsResource = registerResource(Hints).Hints;
+
+    try {
+      await seenHintsResource.get(null, 1);
+    } catch (error) {
+      expect(error.internalError.response).toBeUndefined();
+      expect(error.internalError.message).toBe('Network Error');
+    }
+  });
+});
```
Comment (perprorenata) [inline test-jest/methods.test.js:43]: love this!

**Comment by perjor** [comment]
The only actual thing I added was a
          `internalError: err,`
on the .catch of dispatch in `Rest.js`

Together with a test and a new test suite based on Jest in `test-jest` folder

And decided the confusing
Rest Class inside the Http file
inheriting the Http class from the Methods file confused me for the last time

And Eslint also wanted to be part of this PR

**Review by perprorenata — APPROVED** [review]

File: .babelrc (hunk 1 above)
Comment (perjor) [reply .babelrc:20]: Quite risky. I tried building and tested it in PTE. Seems to still work

File: package.json (hunk 3 above)
Comment (perjor) [reply package.json:27]: Claiming my spot in the wall of fame

File: .babelrc (hunk 1 above)
Comment (perprorenata) [reply .babelrc]: :+1:

**Review by perprorenata — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (perprorenata) [reply package.json:27]: 😁

**Pull request merged**